xbmc/network/upnp/UPnPPlayer.cpp

```diff
@@ -505,49 +509,53 @@ void CUPnPPlayer::Seek(bool bPlus, bool bLargeStep, bool bChapterOverride)
 {
 }
 
-void CUPnPPlayer::DoAudioWork()
+void CUPnPPlayer::Process()
 {
-  NPT_CHECK_POINTER_LABEL_SEVERE(m_delegate, failed);
-  m_delegate->UpdatePositionInfo();
+  while (!m_bStop)
+  {
+    NPT_CHECK_POINTER_LABEL_SEVERE(m_delegate, failed);
+    m_delegate->UpdatePositionInfo();
 
-  if(m_started) {
     NPT_String uri, meta;
     NPT_CHECK_LABEL(m_delegate->m_transport->GetStateVariableValue("CurrentTrackURI", uri), failed);
-    NPT_CHECK_LABEL(m_delegate->m_transport->GetStateVariableValue("CurrentTrackMetadata", meta), failed);
-
-    if(m_current_uri  != (const char*)uri
-    || m_current_meta != (const char*)meta) {
-      m_current_uri  = (const char*)uri;
-      m_current_meta = (const char*)meta;
-      CFileItemPtr item = GetFileItem(uri, meta);
-      g_application.CurrentFileItem() = *item;
-      CServiceBroker::GetAppMessenger()->PostMsg(TMSG_UPDATE_CURRENT_ITEM, 0, -1,
-                                                 static_cast<void*>(new CFileItem(*item)));
-    }
+    NPT_CHECK_LABEL(m_delegate->m_transport->GetStateVariableValue("CurrentTrackMetadata", meta),
+                    failed);
 
-    // Player may be paused or resumed from the target player, state needs to be synchronized to data cache core.
-    CDataCacheCore& dataCacheCore = CDataCacheCore::GetInstance();
-    if (IsPaused() && dataCacheCore.GetSpeed() > 0.0f)
-    {
-      m_callback.OnPlayBackPaused();
-      dataCacheCore.SetSpeed(1.0, 0.0);
-    }
-    else if (!IsPaused() && dataCacheCore.GetSpeed() == 0.0f)
+    if (m_started)
     {
-      m_callback.OnPlayBackResumed();
-      dataCacheCore.SetSpeed(1.0, 1.0);
-    }
+      if (m_current_uri != (const char*)uri || m_current_meta != (const char*)meta)
+      {
+        m_current_uri = (const char*)uri;
+        m_current_meta = (const char*)meta;
```

Comment:
I know you only moved this code, but can we get rid of the c-style casts here?

Reply:
Hum I figured out some hours ago this block of code has 0 chances of being useful at all and actually breaks the artwork since the item is replaced...
So I plan to simply remove this after having a go at fixing playlists. So if you don't mind Id like to keep this as is.

Reply:
Okay.